frontend/src/components/Item/CollectionTabs.vue

```diff
@@ -19,46 +19,33 @@
   </div>
 </template>
 
-<script lang="ts">
+<script setup lang="ts">
 import { BaseItemDto } from '@jellyfin/sdk/lib/generated-client';
 import { groupBy } from 'lodash-es';
-import { defineComponent } from 'vue';
+import { computed, ref, watch } from 'vue';
 import { itemsStore } from '@/store';
 
-export default defineComponent({
-  props: {
-    item: {
-      type: Object as () => BaseItemDto,
-      required: true
-    }
-  },
-  setup() {
-    const items = itemsStore();
+const items = itemsStore();
 
-    return { items };
-  },
-  data() {
-    return {
-      currentTab: 0,
-      loading: false
-    };
-  },
-  computed: {
-    children(): Record<string, BaseItemDto[]> | undefined {
-      return groupBy(this.items.getChildrenOfParent(this.item.Id), 'Type');
+const props = defineProps<{
+  item: BaseItemDto;
+}>();
+
+const currentTab = ref(0);
+const loading = ref(false);
+const children = computed(() => {
+  return groupBy(items.getChildrenOfParent(props.item.Id), 'Type');
+});
+
+watch(
+  [props.item],
```

Comment:
Not preferred, it's the only one that works when watching subproperties.

Reply:
Ah right, I'm learning Vue as I go - I'm used to React with the dependency lists. I did some reading of https://vuejs.org/guide/essentials/watchers.html, and it seems that I need a getter here because props are not reactive by default, and thus a getter is needed to perform comparison. I'll update this to use a getter.

As a quick question, I had a really hard time understanding if props are made reactive by default. My current understanding is no, but they will be reactive if a reactive object is passed in as a prop value. Is that correct?

Reply:
The props item in general is reactive. With watchers, you may need to enable the `deep` attribute (it was on by default iirc, but I think they've changed it) or use a getter. But it may be easier to watch for an attribute, and those aren't reactive unless you pass a reactive object indeed. So for a single property, you need to use a getter like what you did in your latest change :)

Reply:
@ThibaultNocchi @aweebs It wasn't by default before. Read this: https://vuejs.org/guide/essentials/watchers.html#deep-watchers and this: https://v2.vuejs.org/v2/api/#vm-watch

My experience while migrating the app and deep diving the docs tells me that in Vue 3 watchers:

*Assume getters the following syntax: ``() => whatever``*

* Refs and root reactive objects: **Don't need getters**
      * defineProps returns a proxied object with ``reactive``
      * Root reactive objects are basically this: ``const a = reactive(...)``

* Primitives inside reactive objects and reassignment of objects inside reactive objects: **A getter is enough**
* Reassignment inside objects inside reactive objects: **A getter and deep is needed**
     * E.G is not the same to do ``a.b = c`` (being b and c objects) than doing ``a.b.e = false``. In the second one, you're mutating a value inside the nested object, not the nested object itself.